SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -102,7 +109,11 @@ internal void SaveCache()
                 stringBuilder.AppendLine(string.Format("{0}:{1}", entry.Value.Name, entry.Value.Index));
             }
 
+            WaitForFile();
```

Comment:
By the time this is called, `fileInUse` would already be `false`, regardless of whether it's being used or not